packages/synapse-react-client/src/synapse-client/HttpClient.ts

```diff
@@ -32,24 +32,25 @@ const MAX_RETRY = 3
  * @throws SynapseClientError
  */
 const fetchWithExponentialTimeout = async <TResponse>(
-  url: RequestInfo,
+  requestInfo: RequestInfo,
   options: RequestInit,
   delayMs = 1000,
 ): Promise<TResponse> => {
+  const url = typeof requestInfo === 'string' ? requestInfo : requestInfo.url
   let response
   try {
-    response = await fetch(url, options)
+    response = await fetch(requestInfo, options)
   } catch (err) {
     console.error(err)
-    throw new SynapseClientError(0, NETWORK_UNAVAILABLE_MESSAGE, url.toString())
+    throw new SynapseClientError(0, NETWORK_UNAVAILABLE_MESSAGE, url)
```

Comment:
The `url.toString()` was what triggered a new eslint error in the old code because it may not have been a string. With the changes above, `url` is now always a string.